flik/linesearch/conditions.py

```diff
@@ -1 +1,47 @@
 """Conditions for the line search algorithm."""
+def soft_wolfe(gradf, valf, alpha, direction, const2=0.9):
+    """Curvature condition
+
+    Parameters
+    ----------
+    gradf: function
+        Gradient of objective nonlinear function
+    valf: np.ndarray(N,)
+        Function value
+    alpha: float
+        Step length
+    direction: np.ndarray(N,)
+        Descent direction
+    const2: float
+        Constant C2
+
+    Raises
+    ------
+    ValueError
+        If C2 is not between 0 and 1
+    TypeError
+        If gradient function if not a function
+
+    Returns
+    -------
+    Boolean
+        True if curvature condition is satisfied
+    """
+    # Check parameters
+    if not isinstance(valf, (np.ndarray, float)):
+        raise TypeError('Variable should be an array of float')
```

Comment:
If you want it to be an array of float, then you also need to check its data type:
```python
if valf.dtype != float:
    raise TypeError('wrong type')
```
I'm not sure if both float and numpy array should be supported because we might get some errors if we do not explicitly test for float input. For example, line 37 will fail if a float is given (because a float does not have attribute `size`). If it will be supported, then it should be converted into a numpy array for consistency within the code:
```python
if isinstance(valf, float):
    valf = np.array(valf)
elif not isinstance(valf, np.ndarray):
    raise TypeError
```